packages/account/src/Components/additional-kyc-info-modal/additional-kyc-info-form.tsx

```diff
@@ -0,0 +1,247 @@
+        align={isMobile() ? 'left' : 'center'}
+        className='additional-kyc-info-modal__form--header'
+    >
+        <Localize i18n_default_text='Please take a moment to update your information now.' />
+    </Text>
+);
+
+type TAdditionalKycInfoFormProps = {
+    setError?: React.Dispatch<React.SetStateAction<string>>;
+};
+
+export const AdditionalKycInfoForm = observer(({ setError }: TAdditionalKycInfoFormProps) => {
+    const { client, ui, notifications } = useStore();
+    const { residence_list, updateAccountStatus } = client;
+    const {
+        update,
+        mutation: { isLoading, isSuccess, error, isError },
+        data: account_settings,
+        isLoading: isAccountSettingsLoading,
+    } = useSettings();
+
+    const { fields, initialValues, validate } = getFormConfig({
+        account_settings,
+        residence_list,
+        required_fields: ['place_of_birth', 'tax_residence', 'tax_identification_number', 'account_opening_reason'],
+    });
+
+    const onSubmit = (values: typeof initialValues) => {
+        const tax_residence = residence_list.find(item => item.text === values.tax_residence)?.value;
+        const place_of_birth = residence_list.find(item => item.text === values.place_of_birth)?.value;
+
+        update({ ...values, tax_residence, place_of_birth });
+    };
+
+    React.useEffect(() => {
+        if (isSuccess) {
+            updateAccountStatus();
+            notifications.refreshNotifications();
+            ui.toggleAdditionalKycInfoModal();
+            ui.toggleKycInformationSubmittedModal();
+        }
+    }, [isSuccess, notifications, ui, updateAccountStatus]);
+
+    React.useEffect(() => {
+        isError && setError?.(error as string);
+    }, [error, isError, setError]);
+
+    const onItemSelection =
+        (
+            field: string,
+            setFieldValue: (
+                field: string,
+                value: string,
+                shouldValidate?: boolean | undefined
+            ) => Promise<void | FormikErrors<Record<string, unknown>>>
+        ) =>
+        ({ value, text }: TListItem) => {
+            setFieldValue(field, value ? text : '', true);
+        };
+
+    if (isAccountSettingsLoading) {
+        return <Loading is_fullscreen={false} />;
+    }
+
+    return (
+        <Formik
+            validateOnMount
+            validateOnBlur
+            validateOnChange
+            initialValues={initialValues}
+            onSubmit={onSubmit}
+            validate={validate}
+        >
+            {({ isValid, dirty, setFieldValue }) => (
+                <Form className='additional-kyc-info-modal__form-layout'>
+                    {isLoading ? (
+                        <Loading is_fullscreen={false} />
+                    ) : (
+                        <section className='additional-kyc-info-modal__form-layout--fields'>
+                            <FormTitle />
+                            <fieldset className='additional-kyc-info-modal__form-field'>
```

Comment:
@shahzaib-deriv Could we move each field JSX into a separate component? 🤔

Reply:
As stated [here](https://github.com/binary-com/deriv-app/pull/9865#discussion_r1319356659) we'll definitely consider this with a thought of central validation for field which can be used everywhere. As for now we've different forms with different validation rule implementation for the same field